nimbus/src/evaluator.rs

```diff
@@ -45,7 +45,18 @@ pub fn evaluate_enrollment(
     app_context: &AppContext,
     exp: &Experiment,
 ) -> Result<ExperimentEnrollment> {
-    // Get targeting out of the way first - "if let chains" are experimental,
+    // Verify the application-id matches the application being targeted
+    // by the experiment.  This could also be done via the targeting
+    // but this is a short-circuit of sorts since the application field
```

Comment:
I agree this is better than doing it via `targeting` as it would mean an odd refactor - so I suggest you just delete that second sentence from the comment.